src/v2/SuperstateTokenV2.sol

```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import {ERC20Upgradeable} from "openzeppelin-contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import {Ownable2StepUpgradeable} from "openzeppelin-contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {PausableUpgradeable} from "openzeppelin-contracts-upgradeable/security/PausableUpgradeable.sol";
+import {ECDSA} from "openzeppelin-contracts/utils/cryptography/ECDSA.sol";
+
+import {IERC7246} from "src/interfaces/IERC7246.sol";
+import {AllowList} from "src/AllowList.sol";
+
+/**
+ * @title SuperstateToken
+ * @notice A Pausable ERC7246 token contract that interacts with the AllowList contract to check if transfers are allowed
+ * @author Superstate
+ */
+abstract contract SuperstateTokenV2 is ERC20Upgradeable, IERC7246, PausableUpgradeable, Ownable2StepUpgradeable {
+    /// @notice The major version of this contract
+    string public constant VERSION = "2";
+
+    /// @dev The EIP-712 typehash for authorization via permit
+    bytes32 internal constant AUTHORIZATION_TYPEHASH =
+        keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
+
+    /// @dev The EIP-712 typehash for the contract's domain
+    bytes32 internal constant DOMAIN_TYPEHASH =
+        keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");
+
+    /// @notice Admin address with exclusive privileges for minting and burning
+    /// @notice As of v2, this field is no longer used due to implementing Ownable2Step. The field is kept here to keep the memory storage layout consistent.
+    address public immutable admin;
```

Comment:
we cant remove, but we can rename right. maybe "_deprecatedAdmin" or something

Reply:
Addressed in https://github.com/superstateinc/ustb/pull/57